src/results.js

```diff
@@ -31,20 +32,66 @@ class Err {
 	get isErr() { return true }
 	get ok() { return null }
 	get err() { return this._error }
+
 	toString() {
 		if (this._error)
 			return "Error: " + JSON.stringify(this._error)
-		else
-			return "Error"
+		return "Error"
 	}
+
 	toJSON() {
 		let error = (this._error === undefined ? '' : this._error)
 		error = error instanceof Error ? error.toString() : error
 		return { 'Error': error }
 	}
+
+	static _notFound ({ message = 'Not Found', payload, stackTrace } = {}){ 
+		return Err._buildCustomErr('NOT_FOUND', message, payload, stackTrace, 'NotFound', 'notFound')
+	}
+	
+	static _alreadyExists ({ message = 'Entity already exists', payload, stackTrace } = {}) { 
+		return Err._buildCustomErr('ALREADY_EXISTS', message, payload, stackTrace, 'AlreadyExists')
+	}
+	
+	static _invalidEntity ({ message = 'Invalid entity', payload, stackTrace } = {}) { 
+		return Err._buildCustomErr('INVALID_ENTITY', message, payload, stackTrace, 'InvalidEntity')
+	}
+	
+	static _invalidArguments ({ message = 'Invalid arguments', payload, stackTrace } = {}) { 
```

Comment:
Invalid Argument should have a extra argument `argName` to specify the invalid arguments.

Ref: https://docs.microsoft.com/en-us/dotnet/api/system.argumentexception.-ctor?view=net-5.0#System_ArgumentException__ctor_System_String_System_String_